include/seastar/core/app-template.hh

```diff
@@ -106,6 +106,13 @@ public:
         /// You can adjust the behavior of SIGINT/SIGTERM by installing signal handlers
         /// via reactor::handle_signal().
         bool auto_handle_sigint_sigterm = true;
+        /// \brief Reserve -h for the application
+        ///
+        /// Normally, -h is the short-form for --help.
+        /// Some applications however might want to reserve -h for their own use.
+        /// Set this option to true, to reserve -h for the application. In this
+        /// case, --help will not have a short-form option.
+        bool reserve_dash_h = false;
```

Comment:
This flag supports [Avi's concern](https://github.com/scylladb/seastar/pull/1615#issuecomment-1653089501) about design error regarding options parsing by seastar itself. Maybe the better direction is in all-or-nothing approach when app-template refuses to set up any built-in options and lets the application do it

Reply:
Right, this approach doesn't scale. What if another app wants `-c`. Or another wants `-m`.  I think the right approach is another overload of `app_template::run()` which doesn't take `argc` and `argv`, and instead leaves arg parsing to the application.

Reply:
Well, app_template should continue to support the current status-quo (we need it for internal applications like httpd, and it is an easy way to start using). We should also expose the current bpo support as building blocks so an application that wants to continue torturing itself with bpo (or has simply realized there is no good alternative) can continue to do so. But it should be driven by the application, instead of driven by the framework with lots of APIs to adjust the behavior.